spec/requests/completed_learning_units_controller_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+RSpec.describe CompletedLearningUnitsController, type: :request do
+  describe 'POST /create' do
+    let(:user) { create(:user) }
+    let(:curriculum) { create(:curriculum) }
+    let(:learning_unit) { create(:learning_unit) }
+
+    before do
+      create(:curriculum_affiliation, curriculum:, learning_unit:)
+      sign_in user
+    end
+
+    context 'when marking a learning unit as completed' do
+      def perform
+        post learning_unit_completed_learning_units_path(
+          learning_unit_id: learning_unit.id
+        )
+      end
+
+      it 'creates a new record in CompletedLearningUnits' do
+        expect { perform }.to change(CompletedLearningUnit, :count).by(1)
+      end
+
+      it 'redirects to curriculum page' do
+        expect(perform).to redirect_to(
+          root_path
+        )
+      end
+
+      it 'shows a new button with a completed message' do
```

Comment:
I would add a better description to understand the two redirects. Like "following two redirects to the ____ and the ___ it shows a new button with a completed message"